app/middleware/resolve_service.js

```diff
@@ -0,0 +1,23 @@
+const views = require('../utils/views.js')
+const CORRELATION_HEADER = require('../utils/correlation_header.js').CORRELATION_HEADER
+const withAnalyticsError = require('../utils/analytics.js').withAnalyticsError
+const getAdminUsersClient = require('../services/clients/adminusers_client')
+
+module.exports = function (req, res, next) {
+  'use strict'
+  const _views = views.create()
+
+  if (!req.chargeId && !req.chargeData) {
+    _views.display(res, 'UNAUTHORISED', withAnalyticsError())
+  } else {
+    let correlationId = req.headers[CORRELATION_HEADER]
+    return getAdminUsersClient({correlationId: correlationId}).findServiceBy({gatewayAccountId: req.chargeData.gateway_account.gateway_account_id})
```

Comment:
If that's not present that'll be a bigger problem isn't it? Having a charge without a gateway account id? Sounds like a impossible scenario, but I'll surround it using `lodash`.get so at least there won't be referencing errors if that ever happens.